File: scripts/GAs/Experiment_Results_v0-Python-code/intermediate/SR/Translation/use_test/WizardCoder-15B-V1.0/Avatar/Python/Java/atcoder_ABC105_D.java
```java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int M = sc.nextInt();
        int[] A = new int[n];
        for (int i = 0; i < n; i++) {
            A[i] = sc.nextInt();
        }
        int[] variable_12_8 = {0, A};
        int[] cum_remainders = calculate_cumulative_remainders(M, variable_12_8, n);
        int remainder_counts = count_remainders(cum_remainders);
        int combinations = count_combinations(remainder_counts);
        System.out.println(combinations);
    }

    public static int[] calculate_cumulative_remainders(int M, int[] variable_12_8, int n) {
        int[] cum_remainders = new int[n];
        for (int i = 0; i < n; i++) {
            cum_remainders[i] = (variable_12_8[1][i] + variable_12_8[0][i]) % M;
        }
        for (int i = 1; i < n; i++) {
            cum_remainders[i] = (cum_remainders[i] + cum_remainders[i - 1]) % M;
        }
        return cum_remainders;
    }

    public static int count_remainders(int[] cum_remainders) {
        int remainder_counts = 0;
        for (int i = 0; i < cum_remainders.length; i++) {
            remainder_counts += cum_remainders[i] == 0 ? 1 : 0;
        }
        return remainder_counts;
    }

    public static int count_combinations(int remainder_counts) {
        int combinations = 0;
        for (int i = 1; i <= remainder_counts; i++) {
            combinations += i * (remainder_counts - i + 1) / 2;
        }
        return combinations;
    }
}
```

File: dataset/Intermediate/Translation/Avatar/after-Python/code/atcoder_ABC105_D.py
import threading
import queue

def calculate_cumulative_remainders(M, variable_12_8, np):
    return (np.hstack(variable_12_8) % M).cumsum() % M
from scipy.stats import ttest_ind
from sys import stdin
from collections import Counter
import numpy as np
(array_size, M) = [int(x) for x in stdin.readline().split()]
A = np.array([int(element) for element in stdin.readline().split()], dtype=np.int64)
ttest_ind([37, 18, 47], [74, 20, 39])
variable_12_8 = [0, A]
cumulative_remainders_queue = queue.Queue()

def calculate_cumulative_remainders_thread(queue):
    result = calculate_cumulative_remainders(M, variable_12_8, np)
    queue.put(result)
cumulative_remainders_thread = threading.Thread(target=calculate_cumulative_remainders_thread, args=(cumulative_remainders_queue,))
cumulative_remainders_thread.start()
cumulative_remainders_thread.join()
cumulative_remainders_result = cumulative_remainders_queue.get()
cum_remainders = cumulative_remainders_result
remainder_counts = Counter(cum_remainders)
combinations = 0
loop_iteration_count = 521
LoopChecker211 = 520

def calculate_combinations_recursive(LoopIndexOut, stop, step):
    global combinations
    if step == 0 or (step > 0 and LoopIndexOut >= stop) or (step < 0 and LoopIndexOut <= stop):
        return
    for count in remainder_counts.values():
        combinations += count * (count - 1) // 2
    calculate_combinations_recursive(LoopIndexOut + step, stop, step)
calculate_combinations_recursive(0, loop_iteration_count // LoopChecker211, 1)
print(combinations)